File: 003/longest-substring.js
function lengthOfLongestSubstring(s) {
    if (s.length == 0) {
        return 0;
    }
    if (s.length == 1) {
        return 1;
    }
    let barrier = 0;
    let maxLen = 1;
    for (let i = 1; i < s.length; i++) {
        for (let j = i-1; j >= barrier; j--) {
            if (s[i] == s[j]) {
                barrier = j+1;
                break;
            }
        }
        maxLen = Math.max(maxLen, i-barrier+1);
    }
    return maxLen;
}

File: 003/LongestSubstring.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int len = s.length();
        if (len < 2) {
            return len;
        }

        int barrier = 0;
        int maxLen = 1;
        for (int i = 0; i < len; i++) {
            for (int j = i - 1; j >= barrier; j--) {
                if (s.charAt(i) == s.charAt(j)) {
                    barrier = j+1;
                    break;
                }
            }
            maxLen = Math.max(maxLen, i - barrier + 1);
        }
        return maxLen;
    }
}